var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -407,10 +407,14 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     # ###################### Dependencies ##########################
 
     # External Kokkos
-    depends_on("kokkos@4.4.01", when="@master: +kokkos")
-    depends_on("kokkos@4.3.01", when="@16.0.0 +kokkos")
-    depends_on("kokkos@4.2.01", when="@15.1.0:15.1.1 +kokkos")
-    depends_on("kokkos@4.1.00", when="@14.4.0:15.0.0 +kokkos")
+    depends_on("kokkos@4.4.01~complex_align", when="@master: +kokkos")
```

Comment:
Maybe slightly cleaner would be to extract the common when condition, and split the version pinning from general constraints to avoid repetition

```
with when("@14.4: +kokkos"):
    depends_on("kokkos~complex_align")  # generic requirement
    depends_on("kokkos@4.3.0-1", when="@16")  # bunch of version pinning statements
    ...
```